pkg/utils/package.go

```diff
@@ -2,12 +2,56 @@ package utils
 
 import (
 	"fmt"
+	"github.com/mandelsoft/filepath/pkg/filepath"
+	"github.com/mandelsoft/goutils/general"
+	"github.com/mandelsoft/vfs/pkg/osfs"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	"golang.org/x/mod/modfile"
 	"reflect"
 	"runtime"
 	"strings"
 )
 
-const MODULE_PATH = "github.com/open-component-model/ocm"
+func GetRelativePathToProjectRoot(i ...int) (string, error) {
+	iterations := general.OptionalDefaulted(20, i...)
+
+	path := "."
+	for count := 0; count < iterations; count++ {
+		if ok, err := vfs.FileExists(osfs.OsFs, filepath.Join(path, "go.mod")); err != nil || ok {
+			if err != nil {
+				return "", fmt.Errorf("failed to check if go.mod exists: %v", err)
+			}
+			return path, nil
+		}
+		if count == iterations {
+			return "", fmt.Errorf("could not find go.mod (within %d steps)", iterations)
+		}
+		path = filepath.Join(path, "..")
+	}
+	return "", nil
+}
+
+func GetModuleName() (string, error) {
+	pathToRoot, err := GetRelativePathToProjectRoot()
+	if err != nil {
+		return "", err
+	}
+	pathToGoMod := filepath.Join(pathToRoot, "go.mod")
```

Comment:
would it make sense to spend a public const for: go.mod?

```go
const GoMod = "go.mod"
```

Reply:
sure, why not